### ISSUE TITLE
and connected security requirements not possible.

### ISSUE THREAD

**Issue by FalkNisius** [seriousme/fastify-openapi-glue#546]
in the definition of security requirements object is defined, that it can be reference more than one security scheme.
All referenced schemes must be satisfied.

As example two schemes:

```yaml
components:
  securitySchemes:
    apiTenant: 
      type: apiKey
      in: header       
      name: X-ORGANIZATION
    bearerAuth:
      type: http
      scheme: bearer
      bearerFormat: b64token
```

And a path definition using both and connected:

```yaml
paths:
  /example:
    get:
      security:
        - apiTenant: []
          bearerAuth: ['used:scope']
```

The add schemes function in SecurityHandlers fetched only the first found security scheme, and not all.

As workaround it is possible to call the method for the second scheme from the method of the first scheme, but I think it is not intented from the view of the openapi specification.

Thanks for the fantastic work.

**Comment by FalkNisius** [comment]
in the ParserBase the parseSecurity method takes only the first property (`Object.keys(item)[0]`)  and not the complete object. In my opinion the extracted name shoul also be checked against the securitySchemes definition if it exists.

**Comment by FalkNisius** [comment]
Your quote from the spec is right, but the wrong part. If I specify 

```yaml
  security:
     - apiTenant: []
     - bearerAuth: ["some:scope"]
```

You are complete right, and the plugin does what it's supposed to do.

If I specify

```yaml
  security:
     - apiTenant: []
       bearerAuth: ["some:scope"]
```
or in i a more JSON style

```yaml
  security: [{"apiTenant": [], "bearerAuth": ["some:scope"]}]
```

Than the security requirement object contains two schemes `apiTenant` and `bearerAuth` and both are combined with AND. I quote the paragraph above at the same place in the spec.

> Security Requirement Objects that contain multiple schemes require that all schemes MUST be satisfied for a request to be authorized. This enables support for scenarios where multiple query parameters or HTTP headers are required to convey security information.

The parser fetched only the first scheme of the security requirements object.

Doe's this help?

Kind Regards,
Falk

**Comment by seriousme** [comment]
Yes it does, thanks for highlighting ! 
Sounds indeed like a bug, I will look into it.

Kind regards,
Hans

**Comment by seriousme** [comment]
The fix is in https://github.com/seriousme/fastify-openapi-glue/pull/548 which results in more clean code for this functionality.  Thanks for raising the issue!

Released as 4.4.3 on NPM.

Can you please check your use case ?

Kind regards,
Hans

**Comment by seriousme** [comment]
Closing the issue now, feel free to reopen or open a new issue should you encounter anything else.

Kind regards,
Hans
